spec/solver_spec.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'spec_helper'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  context 'To test the solver class' do
+    it 'Returns the factorial of the number' do
+      expect(@solver.factorial(5)).to eq 120
+    end
+
+    it 'Reverses the String input' do
+      expect(@solver.reverse('abby')).to eq 'ybba'
+    end
+
+    it 'Runs the fizzbuzz algorithm' do
+      expect(@solver.fizzbuzz(15)).to eq 'FizzBuzz'
+    end
```

Comment:
- [ ] Also consider adding test cases that handle numbers divisible by 3 and 5, also for numbers not divisible by 3 or 5 like 7

Reply:
Added the additional test cases